broadinstitute/wfl
# [GH-1540] Apply User Comments to Terra Submissions

### Purpose

We want to be able to utilize the user comment field in a Terra submission. In this comment, we would like to include the information necessary to be able to commence a retry such as workload uuid and environment. In order to accomplish this, WFL also needs to be updated to have the workload available for the submission creation to be able to send a user comment.

- (https://broadinstitute.atlassian.net/browse/GH-1540)

### Changes


- Added user comment to submission creation
- Updated the three workload update functions (source, executor, sink) to be passed a workload instead of their object and upstream from the workload.
- Updated tests to accommodate

### Review Instructions


- Run tests
- Verify that a user comment with the workload uuid and the current running environment makes it into a Terra submission
